backend/papersfeed/utils/replies/utils.py

```diff
@@ -0,0 +1,325 @@
+
+    # Text
+    text = args[constants.TEXT]
+
+    # Check parameter
+    if not text:
+        raise ApiError(constants.PARAMETER_ERROR)
+
+    # Check collection id
+    if not Collection.objects.filter(id=collection_id).exists():
+        raise ApiError(constants.NOT_EXIST_OBJECT)
+
+    # Create
+
+    reply = Reply.objects.create(
+        user_id=request_user.id,
+        text=text,
+    )
+
+    ReplyCollection.objects.create(
+        collection_id=collection_id,
+        reply_id=reply.id
+    )
+
+    replies, _, _ = __get_replies(Q(id=reply.id), request_user, None)
+
+    if not replies:
+        raise ApiError(constants.NOT_EXIST_OBJECT)
+
+    reply = replies[0]
+
+    return reply
+
+def insert_reply_review(args):
+    """Insert Reply Review"""
+    is_parameter_exists([
+        constants.ID, constants.TEXT
+    ], args)
+
+    # Review Id
+    review_id = args[constants.ID]
+
+    # Request User
+    request_user = args[constants.USER]
+
+    # Text
+    text = args[constants.TEXT]
+
+    # Check parameter
+    if not text:
+        raise ApiError(constants.PARAMETER_ERROR)
+
+    # Check review id
+    if not Review.objects.filter(id=review_id).exists():
+        raise ApiError(constants.NOT_EXIST_OBJECT)
+
+    # Create
+    reply = Reply.objects.create(
+        user_id=request_user.id,
+        text=text
+    )
+
+    ReplyReview.objects.create(
+        review_id=review_id,
+        reply_id=reply.id
+    )
+
+    replies, _, _ = __get_replies(Q(id=reply.id), request_user, None)
+
+    if not replies:
+        raise ApiError(constants.NOT_EXIST_OBJECT)
+
+    reply = replies[0]
+
+    return reply
+
+def update_reply(args):
+    """Update Reply"""
+    is_parameter_exists([
+        constants.ID
+    ], args)
+
+    # Reply Id
+    reply_id = args[constants.ID]
+
+    # Request User
+    request_user = args[constants.USER]
+
+    try:
+        reply = Reply.objects.get(id=reply_id)
+    except ObjectDoesNotExist:
+        raise ApiError(constants.NOT_EXIST_OBJECT)
+
+    # Auth
+    if reply.user_id != request_user.id:
+        raise ApiError(constants.AUTH_ERROR)
+
+    # Text
+    text = args[constants.TEXT] if constants.TEXT in args else None
+
+    # Update Text
+    if text:
+        reply.text = text
+
+    reply.save()
+
+    replies, _, _ = __get_replies(Q(id=reply.id), request_user, None)
+
+    if not replies:
+        raise ApiError(constants.NOT_EXIST_OBJECT)
+
+    reply = replies[0]
+
+    return reply
+
+def remove_reply(args):
+    """Remove Reply"""
+    is_parameter_exists([
+        constants.ID
+    ], args)
+
+    # Reply ID
+    reply_id = args[constants.ID]
+
+    # Request User
+    request_user = args[constants.USER]
+
+    try:
+        reply = Reply.objects.get(id=reply_id)
+    except ObjectDoesNotExist:
+        raise ApiError(constants.NOT_EXIST_OBJECT)
+
+    # Auth
+    if reply.user_id != request_user.id:
+        raise ApiError(constants.AUTH_ERROR)
+
+    reply.delete()
```

Comment:
It doesn’t return anything. I think the related packing in apis.py is unnecessary.

Reply:
Oh thank you!